gax-grpc/src/main/java/com/google/api/gax/grpc/OperationFuture.java

```diff
@@ -29,348 +29,58 @@
  */
 package com.google.api.gax.grpc;
 
-import com.google.api.core.AbstractApiFuture;
 import com.google.api.core.ApiFuture;
 import com.google.api.core.BetaApi;
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.util.concurrent.SettableFuture;
 import com.google.longrunning.Operation;
-import com.google.longrunning.OperationsClient;
-import com.google.protobuf.Any;
-import com.google.protobuf.InvalidProtocolBufferException;
 import com.google.protobuf.Message;
-import io.grpc.Status;
-import java.util.concurrent.Callable;
-import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Executor;
-import java.util.concurrent.Future;
-import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
-import org.threeten.bp.Duration;
 
 /** An ApiFuture which polls a service through OperationsApi for the completion of an operation. */
 @BetaApi
-public final class OperationFuture<ResponseT extends Message> extends AbstractApiFuture<ResponseT> {
-  static final Duration DEFAULT_POLLING_INTERVAL = Duration.ofSeconds(1);
+public final class OperationFuture<ResponseT extends Message> implements ApiFuture<ResponseT> {
```

Comment:
There is a way:
`operationFuture.getInitialFuture().get().getMetadata()` for metadata, and `operationFuture.getInitialFuture().get().getName()` for name.

I can add the wrapping methods, though they will be blocking methods. It feels not intuitive for a user: 1) why the methods are blocking and 2) (more importantly) on what they are blocking, and when a user can expect those to unblock. To answer the second quesition a user needs to know about existence of the initial future, so I decided just give him the instance explicitly, so he is not confused and also has more power (like adding a listener to it).